[DOC-9307] Publish v23.2.0-beta.2 release notes (#18129)
cockroachdb/docs: src/current/_includes/releases/v23.2/v23.2.0-beta.2.md

<h3 id="v23-2-0-beta-2-general-changes">General changes</h3>

- CockroachDB now periodically dumps the state of its internal memory accounting system into the `heap_profiler/` directory when a heap profile is taken. To disable this behavior, set the `diagnostics.memory_monitoring_dumps.enabled` [cluster setting](https://cockroachlabs.com/docs/v23.2/cluster-settings) to `false`. [#114998][#114998]
- Multi-level compactions have been disabled to investigate possible performance issues with foreground throughput and latency. [#115481][#115481]

<h3 id="v23-2-0-beta-2-enterprise-edition-changes">Enterprise Edition changes</h3>

- When using [Physical Cluster Replication](https://cockroachlabs.com/docs/v23.2/physical-cluster-replication-overview), you can now [initiate a cutover](https://cockroachlabs.com/docs/v23.2/cutover-replication) as of `LATEST` before the initial scan completes. [#115101][#115101]

> - Sensitive information such as `api_secret`, `sasl_password`, `client_cert`, and `ca_cert`, is now redacted in output from commands `SHOW CHANGEFEED JOB`, `SHOW CHANGEFEED JOBS`, and `SHOW JOBS`. [#115567][#115567]

Reviewer: ```suggestion
- Sensitive information such as `api_secret`, `sasl_password`, `client_cert`, and `ca_cert`, is now redacted in output from commands `SHOW CHANGEFEED JOB`, `SHOW CHANGEFEED JOBS`, and [`SHOW JOBS`](https://cockroachlabs.com/docs/v23.2/show-jobs). [#115567][#115567]
```

> - The `physical_replication.frontier_lag_nanos` metric and the related DB Console graph have been removed because they sometimes display incorrect information. For alerting, it is recommended to use the new metric `physical_replication.replicated_time_seconds` metric instead. [#115234][#115234]

Reviewer: ```suggestion
- The `physical_replication.frontier_lag_nanos` metric and the related DB Console graph have been removed because they sometimes display incorrect information. For [alerting](https://cockroachlabs.com/docs/v23.2/physical-cluster-replication-monitoring#prometheus), it is recommended to use the new metric `physical_replication.replicated_time_seconds` metric instead. [#115234][#115234]
```

> - Fixes a bug in physical cluster replication where replicating from a source cluster that is on a version prior to v23.2.x to a destination cluster running on v23.2.x could fail because of an undefined builtin function in the source cluster. [#114257][#114257]

Reviewer: ```suggestion
- Fixed a bug in [physical cluster replication](https://cockroachlabs.com/docs/v23.2/physical-cluster-replication-overview) where replicating from a primary cluster that is on a version prior to v23.2.x to a standby cluster running on v23.2.x could fail because of an undefined builtin function in the primary cluster. [#114257][#114257]
```

<h3 id="v23-2-0-beta-2-db-console-changes">DB Console changes</h3>

- In the [Changeeds dashboard](https://www.cockroachlabs.com/docs/v23.2/ui-cdc-dashboard), the **Max Checkpoint Latency** chart title now refers to "Lag" rather than "Latency", to better reflect the intention of the underlying metric, which measures how recently the changefeed was last checkpointed. [#115003][#115003]
- Times on the X-Axis of bar charts in **Statement details** pages are now correctly formatted in UTC. [#115220][#115220]
- In the **SQL Activity** **Transaction Details** page, you can now view a transaction fingerprint ID across multiple applications by specifying the application name name in the `appNames` URL `GET` parameter using a comma separated encoded string of transaction fingerprint IDs. [#115204][#115204]

<h3 id="v23-2-0-beta-2-bug-fixes">Bug fixes</h3>

> - Fixed a bug that prevented the **Now** button on time range selectors from working as expected when a custom time period was previously selected. [#115514][#115514]

Reviewer: ```suggestion
- Fixed a bug that prevented the **Now** button on time range selectors in the DB Console from working as expected when a custom time period was previously selected. [#115514][#115514]
```

- Fixed a bug introduced in v23.1 that could cause an internal error when using the text format (as opposed to binary) when [preparing a statement](https://www.cockroachlabs.com/docs/v23.2/sql-grammar#prepare_stmt) with a user-defined composite type. [#115064][#115064]
- Fixed a bug that could cause a replica to be stuck processing in a queue's replica set when the replica had recently been removed from purgatory for processing but was destroyed, or the replica's ID changed before being processed. These replicas are now removed from the queue when they are encountered. [#115037][#115037]
- Fixed a bug that could cause a [prepared statement](https://www.cockroachlabs.com/docs/v23.2/sql-grammar#prepare_stmt) to fail if it references both an `enum` and a table that has undergone a schema change. [#115132][#115132]
- Fixed a bug that could cause cluster version finalization to contend with descriptor lease renewals on large clusters. Descriptor lease renewals previously had a higher priority than cluster upgrade finalization. Finalization now always has a higher priority than descriptor lease renewal. [#115034][#115034]

> - Fixed a bug that prevented backups from distributing work evenly across all replicas, including followers, regardless of leaseholder placement. [#115019][#115019]

Reviewer: ```suggestion
- Fixed a bug that prevented [backups](https://cockroachlabs.com/docs/v23.2/backup) from distributing work evenly across all replicas, including followers, regardless of leaseholder placement. [#115019][#115019]
```

<h3 id="v23-2-0-beta-2-general-changes">General changes</h3>

- CockroachDB now periodically dumps the state of its internal memory accounting system into the `heap_profiler/` directory when a heap profile is taken. To disable this behavior, set the `diagnostics.memory_monitoring_dumps.enabled` [cluster setting](https://cockroachlabs.com/docs/v23.2/cluster-settings) to `false`. [#114998][#114998]
- Multi-level compactions have been disabled to investigate possible performance issues with foreground throughput and latency. [#115481][#115481]

<h3 id="v23-2-0-beta-2-enterprise-edition-changes">Enterprise Edition changes</h3>

- When using [Physical Cluster Replication](https://cockroachlabs.com/docs/v23.2/physical-cluster-replication-overview), you can now [initiate a cutover](https://cockroachlabs.com/docs/v23.2/cutover-replication) as of `LATEST` before the initial scan completes. [#115101][#115101]

> - Sensitive information such as `api_secret`, `sasl_password`, `client_cert`, and `ca_cert`, is now redacted in output from commands `SHOW CHANGEFEED JOB``, `SHOW CHANGEFEED JOBS``, and `SHOW JOBS`. [#115567][#115567]

Author: ```suggestion
- Sensitive information such as `api_secret`, `sasl_password`, `client_cert`, and `ca_cert`, is now redacted in output from commands `SHOW CHANGEFEED JOB`, `SHOW CHANGEFEED JOBS`, and `SHOW JOBS`. [#115567][#115567]
```

> - Sensitive information such as `api_secret`, `sasl_password`, `client_cert`, and `ca_cert`, is now redacted in output from commands `SHOW CHANGEFEED JOB``, `SHOW CHANGEFEED JOBS``, and `SHOW JOBS`. [#115567][#115567]

Reviewer: ```suggestion
- Sensitive information such as `api_secret`, `sasl_password`, `client_cert`, and `ca_cert`, is now redacted in output from commands `SHOW CHANGEFEED JOB`, `SHOW CHANGEFEED JOBS`, and [`SHOW JOBS`](https://cockroachlabs.com/docs/v23.2/show-jobs). [#115567][#115567]
```

> - Sensitive information such as `api_secret`, `sasl_password`, `client_cert`, and `ca_cert`, is now redacted in output from commands `SHOW CHANGEFEED JOB``, `SHOW CHANGEFEED JOBS``, and `SHOW JOBS`. [#115567][#115567]

Author: ```suggestion
- Sensitive information such as `api_secret`, `sasl_password`, `client_cert`, and `ca_cert`, is now redacted in output from commands `SHOW CHANGEFEED JOB`, `SHOW CHANGEFEED JOBS`, and [`SHOW JOBS`](https://cockroachlabs.com/docs/v23.2/show-jobs). [#115567][#115567]
```

<h3 id="v23-2-0-beta-2-db-console-changes">DB Console changes</h3>

- In the [Changeeds dashboard](https://www.cockroachlabs.com/docs/v23.2/ui-cdc-dashboard), the **Max Checkpoint Latency** chart title now refers to "Lag" rather than "Latency", to better reflect the intention of the underlying metric, which measures how recently the changefeed was last checkpointed. [#115003][#115003]
- Times on the X-Axis of bar charts in **Statement details** pages are now correctly formatted in UTC. [#115220][#115220]
- The `physical_replication.frontier_lag_nanos` metric and the related DB Console graph have been removed because they sometimes display incorrect information. For alerting, it is recommended to use the new metric `physical_replication.replicated_time_seconds` metric instead. [#115234][#115234]

> - In the **SQL Activity** **Transaction Details** page, you can now view a transaction fingerprint ID across multiple applications by specifying the application name name in the `appNames` URL `GET` parameter using a comma separated encoded string of transaction fingerprint IDs. [#115204][#115204]

Reviewer: ```suggestion
- In the **SQL Activity** **Transaction Details** page, you can now view a transaction fingerprint ID across multiple applications by specifying the application name in the `appNames` URL `GET` parameter using a comma-separated encoded string of transaction fingerprint IDs. [#115204][#115204]
```
